src/Extractor.ts

```diff
@@ -1047,32 +1094,43 @@ class Extractor {
     if (name == null) return null;
 
     const description = this.collectDescription(node);
-    this.recordTypeName(node, name, "INPUT_OBJECT");
 
-    let fields: InputValueDefinitionNode[] | null = null;
+    if (
+      node.type.kind === ts.SyntaxKind.TypeReference &&
+      this.hasTag(node, EXTERNAL_TAG)
```

Comment:
What is the user experience if they use `@gqlExternal` but it's not a type reference? Does it just silently ignore the `@gqlExternal`? If so, can we avoid that with a custom error message for that case?